### PR TITLE
[Input] Class to FC

### LINKED ISSUE #3029: Element type is invalid when using FC Component in inputComponent of <Input/>
Type '() => JSX.Element' is not assignable to type 'typeof Component'.
  Type '() => Element' provides no match for the signature 'new <P = {}, S = {}, SS = any>(props: P | Readonly<P>): Component<P, S, SS>'.ts(2322)
Input.d.ts(45, 5): The expected type comes from property 'InputComponent' which is declared here on type 'IntrinsicAttributes & Omit<TextInputProps & RefAttributes<TextInput> & { containerStyle?: StyleProp<ViewStyle>; ... 15 more ...; renderErrorMessage?: boolean | undefined; } & Partial<...>, keyof ThemeProps<...>> & ... 4 more ... & Partial<...>'

const RadioGroup: FC<IProps> = ({
    values = [
        {key: '1', label: 'male'},
        {key: '2', label: 'famale'},
    ],
    initialValues,
    onChange,
}: IProps) => {
    return (
        <View style={styles.radioGroupCon}>
            {values.map(item => (
                <CheckBox
                    center
                    containerStyle={styles.containerStyle}
                    title={item.label}
                    checkedIcon="dot-circle-o"
                    uncheckedIcon="circle-o"
                    checked={item.key === initialValues}
                    onPress={() => {
                        onChange(item.key);
                    }}
                />
            ))}
        </View>
    );
};

export default RadioGroup;

### PR DESCRIPTION
**Pull request by arpitBhalla**
**What kind of change does this PR introduce?**
Convert Class based `Input` Component to `Functional Component`.

**Did you add tests for your changes?**
Yes, I have updated the tests and migrated them to RN-testing Library.

**If relevant, did you update the documentation?**
No changes are required.

**Does this PR introduce a breaking change?**
No, this PR would not break anything.

**Other information**
Closes #3029
GSoC Task of @arpitBhalla.

- [ ] Self Review for any kind of error in code.

#### I have tested the PR manually in demo app / Snack on 
- [ ] Android
- [ ] iOS
- [ ] Web

### CODE REVIEW

File: src/Input/__tests__/Input.test.tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+      //   const focus = jest.fn();
+      //   const ref = React.createRef<TextInput>();
+      //   render(<TextInput ref={ref} onFocus={focus} />, {
+      //     createNodeMock: (element) => ({
+      //       focus,
+      //     }),
+      //   });
+      //   ref.current.focus();
+      //   expect(focus).toHaveBeenCalledTimes(1);
+      // });
```
Comment (arpitBhalla) [inline src/Input/__tests__/Input.test.tsx:188]: @pranshuchittora how to write tests for refs?

**Comment by arpitBhalla** [comment]
@flyingcircle @pranshuchittora Kindly Review

**Comment by flyingcircle** [comment]
Hitting an error here too

```
ReferenceError: Can't find variable: React
```

**Comment by arpitBhalla** [comment]
```
ReferenceError: React is not defined
ThemeProvider.render
node_modules/react-native-elements/dist/config/ThemeProvider.js:59
```
I guess it's show after merging `next` not due to this PR

**Comment by arpitBhalla** [comment]
```
ReferenceError: React is not defined
ThemeProvider.render
node_modules/react-native-elements/dist/config/ThemeProvider.js:59
```
I guess it's due to merging `next`.

**Comment by arpitBhalla** [comment]
https://github.com/react-native-elements/react-native-elements/commit/687c699915928f2763da73f42c7dfc4aaa1aea7d#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0

**Comment by arpitBhalla** [comment]
Here's change:-https://github.com/react-native-elements/react-native-elements/commit/687c699915928f2763da73f42c7dfc4aaa1aea7d#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0

**Pull request closed without merging**